lang/rs/bindings/cners/examples/loopback/src/main.rs

```diff
@@ -0,0 +1,221 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright (c) 2020-2022 Intel Corporation.
+ */
+
+use clap::{App, Arg};
+use crossbeam_channel::{bounded, select, tick, Receiver};
+use ctrlc;
+use etherparse::Ethernet2Header;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+use std::thread;
+use std::time::Duration;
+
+use cners::error::*;
+use cners::instance::*;
+use cners::packet::*;
+use cners::port::*;
+
+fn main() {
+    // Parse command line arguments.
+    let matches = App::new("loopback")
+        .version("0.1.0")
+        .about("CNE loopback example application.")
+        .arg(
+            Arg::with_name("config")
+                .short("c")
+                .long("config")
+                .required(true)
+                .takes_value(true)
+                .help("CNE JSON config file"),
+        )
+        .arg(
+            Arg::with_name("port")
+                .short("p")
+                .long("port")
+                .required(true)
+                .takes_value(true)
+                .help("port name in JSON file"),
+        )
+        .arg(
+            Arg::with_name("affinity")
+                .short("a")
+                .long("affinity")
+                .required(false) // optional parameter
+                .takes_value(true)
+                .allow_hyphen_values(true) // allow negative values
+                .help("Core affinity for loopback thread. If provided, can help to improve performance"),
+        )
+        .get_matches();
+
+    // start logging
+    env_logger::builder()
+        .try_init()
+        .expect("Failed to initialize event logger");
+
+    log::info!("CNE loopback application. Press ctrl-c to exit");
+
+    // Parse config file.
+    let jsonc_file = matches
+        .value_of("config")
+        .unwrap()
+        .parse::<String>()
+        .expect("Invalid CNE JSON file name");
+
+    // Parse port name.
+    let port_name = matches
+        .value_of("port")
+        .unwrap()
+        .parse::<String>()
+        .expect("Invalid port name");
+
+    let mut core_id = -1;
+    if matches.is_present("affinity") {
+        core_id = matches
+            .value_of("affinity")
+            .unwrap()
+            .parse::<i16>()
+            .expect("Invalid core id");
+    }
+
+    // Get CNE instance.
+    let cne = CneInstance::get_instance();
+
+    // Configure CNE.
+    match cne.configure(&jsonc_file) {
+        Err(e) => {
+            log::error!("Error configuring CNE: {}", e.to_string());
+            return;
+        }
+        Ok(_) => {}
+    }
+
+    // Get CNE port to recv/send packets.
+    let port = cne.get_port(&port_name);
```

Comment:
We should not be referencing ports by name per say and all of the CNDP APIs use port number. I assume the port here is just a number from 0 - N.

The ports names in the application should not be used other than printing out information. This way we do not need to update the command line or hardcode the names in the application.

Reply:
same as https://github.com/CloudNativeDataPlane/cndp/pull/17#discussion_r859992279